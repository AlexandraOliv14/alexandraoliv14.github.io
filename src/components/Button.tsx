import React from 'react'

interface Prop {
    text: string;
    color?: string;
    onClick: ()=>void;
}

export const Button = ({ text = '', color, onClick }: Prop) => {
    return (
        <button className='button_nav' onClick={onClick} style={{ background: color }}>{text}</button>
    )
}
