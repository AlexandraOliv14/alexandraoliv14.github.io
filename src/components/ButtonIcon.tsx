import React from 'react'

interface Props{
    url?:string;
    href:string;
    children?: JSX.Element[] | JSX.Element;
    color?:string;
}
export const ButtonIcon = ({url,href, color,children}:Props) => {
  return (
    <div className='button_icon' style={{background:color}}>
        <a href={href} target="_blank" rel="noopener noreferrer">
            {children}
        </a>
    </div>
  )
}
