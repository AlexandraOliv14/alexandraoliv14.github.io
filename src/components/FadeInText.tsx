
interface Props{
    text:string;
}

export const FadeInText = ({ text }:Props) => {
    return (
        <div className="p-container">
            <div className="p">{addAnimation(text)}</div>
        </div>
    );
};

const addAnimation = (text:any) => {
    const ANIMATION_DELAY = 6;
    const allChars = [...text].map((char) => (char === " " ? <>&nbsp;</> : char));
    
    return allChars.map((char, i) => {
        return (
            <span
                key={i}
                className="char"
                style={{ animationDelay: `${ANIMATION_DELAY * i}ms` }}
            >
                {char}
            </span>
        );
    });
};
