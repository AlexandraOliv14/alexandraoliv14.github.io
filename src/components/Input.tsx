import React from 'react'

interface Props {
    onChange: (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement | HTMLSelectElement>) => void;
    value: any;
    label: string;
    name: string;
}
export const Input = ({ onChange, value, name, label }: Props) => {
    return (
        <div className="name_ii_Active">
            <div className="label">{label}</div>
            <input onChange={onChange} value={value} name={name} className="innerframe" />
        </div>
    )
}
