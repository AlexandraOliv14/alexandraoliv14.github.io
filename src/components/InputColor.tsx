import React from 'react'

interface Props {
    onChange: (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement | HTMLSelectElement>) => void;
    value: any;
    label: string;
    name: string;
}

export const InputColor = ({ onChange, value, name, label }: Props) => {
    return (
        <div className='button-color'>
            <div className="label">{label}</div>
            <div className="color-input-wrapper">
                <input type="color" onChange={onChange} value={value} name={name}/>
            </div>
        </div>
    )
}
