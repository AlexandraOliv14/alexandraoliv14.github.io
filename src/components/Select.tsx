import React from 'react'

interface Props {
    onChange: (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement | HTMLSelectElement>) => void;
    value: any;
    label: string;
    name: string;
    opcions: (string | number)[];
}

export const Select = ({ onChange, value, name,label, opcions }: Props) => {
    return (
        <div className="name_ii_Active">
            <div className="label">{label}</div>
            <select onChange={onChange} value={value} name={name} className="innerframe">
                {
                    opcions.map((value, index) => (
                        <option key={index} value={value}>{value}</option>
                    ))
                }
            </select>
        </div>
    )
}
