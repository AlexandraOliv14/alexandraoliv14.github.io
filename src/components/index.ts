export * from './Button';
export * from './ButtonIcon';
export * from './FadeInText';
export * from './Input';
export * from './InputColor';
export * from './Select';
export * from './Separacion';
