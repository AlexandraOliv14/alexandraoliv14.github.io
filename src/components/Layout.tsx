import React from 'react'
import { Outlet, useNavigate } from 'react-router-dom'
import { Button } from './Button';
// FiPackage ->proyecto

export const Layout = () => {

    const navigate = useNavigate();

    const handlerClick = (link: string) => {
        navigate(link);
    }
    return (
        <>
            <nav className='nav'>
                <div className='titulo'>
                    <h1>Alexandra Olivares SolÃ­s</h1>
                </div>
                <div className='navegacion'>
                    <Button text='Home' onClick={() => handlerClick('/')} />
                    <Button text='About' onClick={() => handlerClick('/about')} />
                    <Button text='Prototipo' onClick={() => handlerClick('/about')} />
                </div>
            </nav>
            <Outlet />
        </>
    )
}