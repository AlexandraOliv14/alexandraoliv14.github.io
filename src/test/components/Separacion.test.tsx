import { render } from "@testing-library/react"
import { Separacion } from "../../components";

describe('Pruebas en <Separacion/>', () => {
    
    test('Separacion debe hacer match con snapshot',()=>{
            const {container} = render(<Separacion/>);
            expect(container).toMatchSnapshot();
        })
})