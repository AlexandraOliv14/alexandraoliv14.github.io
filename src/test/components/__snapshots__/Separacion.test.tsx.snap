// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Separacion/> Separacion debe hacer match con snapshot 1`] = `
<div>
  <div
    class="separacion"
  />
</div>
`;
